curated_transformers/models/with_strided_spans.py

```diff
@@ -51,25 +51,65 @@ def with_strided_spans_forward(
 
     Y_layer, backprop_layer = model.layers[0](spans, is_train=is_train)
 
+    _average_overlaps(model, Y_layer, doc_lens, stride=stride, window=window)
+    Y_docs = _strided_arrays_to_ragged(
+        model, Y_layer, doc_lens, stride=stride, window=window
+    )
+
     def backprop(dY):
         dY_spans, dY_lengths = _ragged_to_strided_arrays(
             dY, stride=stride, window=window
         )
         dXlf = backprop_layer(dY_spans)
+        _average_overlaps(model, Y_layer, doc_lens, stride=stride, window=window)
```

Comment:
Still in draft. Because I am not quite sure yet if we want to do this in backprop. Plus the averaging should probably happen before backprop'ing into the wrapped layer?

Reply:
By the way, I just realized that we are backproping for the overlapping outputs twice? `window - stride` `dY`s are used for both the end of the current window and the start of the next window. I am not sure whether this is fine or not?

Reply:
We discussed this a bit and I greed it would be best to use the mathematically-correct (?) approach of backprop'ing twice with half the gradient (since on the forward we do `(u + v) / 2`). Empirically there doesn't seem to be much of a difference between different gradients, as long as we are averaging the overlaps.